pkg/controller/che/k8s_helpers.go

```diff
@@ -269,16 +272,61 @@ func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, url string)
 		// sometimes timing conditions apply, and host isn't available right away
 		if len(testRoute.Spec.Host) < 1 {
 			time.Sleep(time.Duration(1) * time.Second)
-			testRoute := r.GetEffectiveRoute(instance, "test")
-			requestURL = "https://" + testRoute.Spec.Host
+			testRoute = r.GetEffectiveRoute(instance, "test")
 		}
 		requestURL = "https://" + testRoute.Spec.Host
 
 	} else {
-		requestURL = url
+		requestURL = endpointUrl
+	}
+
+	//adding the proxy settings to the Transport object
+	transport := &http.Transport{}
+
+	if instance.Spec.Server.ProxyURL != "" {
```

Comment:
You're adding +40 lines in already huge func, what you think about extracting some parts in a dedicated func like https://github.com/eclipse/che-operator/commit/dec974f0429215b4731213a6a972000b1e8d1150

^ this is just proposal and does not block this PR

Reply:
well, the thing is that I already tested this code in a disconnected environment. in both the upstream and CRW cases. And I'd have to restart disconnected env configuration + 2 tests in order to test the refactoring.
Would you mind opening a issue to track:
- the cherry-pick of this to the master branch
- the addition, when cherry-picking to master, of the function extraction you proposed ?

Reply:
I'm OK with it since it's an urgent fix which we need to deliver ASAP